src/sources/http_client/client.rs

```diff
@@ -250,13 +250,13 @@ impl Query {
                     if !compilation_result.warnings.is_empty() {
                         let warnings =
                             format_vrl_diagnostics(param.value(), compilation_result.warnings);
-                        warn!(message = "VRL compilation warnings.", %warnings, internal_log_rate_limit = true);
+                        warn!(message = "VRL compilation warnings.", %warnings);
                     }
                     Some(compilation_result.program)
                 }
                 Err(diagnostics) => {
                     let error = format_vrl_diagnostics(param.value(), diagnostics);
-                    warn!(message = "VRL compilation failed.", %error, internal_log_rate_limit = true);
+                    warn!(message = "VRL compilation failed.", %error);
```

Comment:
Unrelated: I think we should bail here instead of warning and continuing. This hides nasty bugs